Classes/DataProcessing/LanguageMenuProcessor.php

```diff
@@ -0,0 +1,333 @@
+<?php
+
+/*
+ * This file is part of the package bk2k/bootstrap-package.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace BK2K\BootstrapPackage\DataProcessing;
+
+use BK2K\BootstrapPackage\Utility\LanguageUtility;
+use TYPO3\CMS\Frontend\DataProcessing\MenuProcessor;
+
+/**
+ * This menu processor utilizes HMENU to generate a json encoded menu
+ * string that will be decoded again and assigned to FLUIDTEMPLATE as
+ * variable. Additional DataProcessing is supported and will be applied
+ * to each record.
+ *
+ * Options:
+ * as - The variable to be used within the result
+ * levels - Number of levels of the menu
+ * expandAll = If false, submenus will only render if the parent page is active
+ * includeSpacer = If true, pagetype spacer will be included in the menu
+ * titleField = Field that should be used for the title
+ *
+ * See HMENU docs for more options.
+ * https://docs.typo3.org/typo3cms/TyposcriptReference/ContentObjects/Hmenu/Index.html
+ *
+ *
+ * Example TypoScript configuration:
+ *
+ * 10 = BK2K\BootstrapPackage\DataProcessing\LanguageMenuProcessor
+ * 10 {
+ *   as = languagenavigation
+ *   expandAll = 1
+ *   includeSpacer = 1
+ *   dataProcessing {
+ *     10 = TYPO3\CMS\Frontend\DataProcessing\FilesProcessor
+ *     10 {
+ *        references.fieldName = media
+ *     }
+ *   }
+ * }
+ */
+class LanguageMenuProcessor extends MenuProcessor
```

Comment:
We should not depend on the currently existing menu processor, i will cause problems to keep them compatible and we do not need a lot of the stuff the menu processor does. I would suppose in that case to duplicate the code and remove everything we do not need.

Reply:
Currently not sure about `localizedTitle` because as user i would expect to find my language wether in english or in my local language.

That would mean if i am browsing a russian website in russian - i would like to have `german` or `deutsch` in the language selector because i propably have no idea that i would be `немецкий`

Reply:
Output contains still the property `spacer` that is not usefull when handling language menus. Should be reomoved.

Reply:
to clarify my idea about the titles also told to benni and he thinks also about this, but no feedback so far: the title is used primary for the be, so in editors language or english. if this differs from the fe user language what's normally the case the editor can set the nav_title with the destination language of the fe user. so it should be exactly what you wrote. have a look at this https://github.com/gilbertsoft/typo3-sites/blob/fdf7713f871637752dc1816aa2c0d0b307fe9676/Configuration/SiteConfigurationTCA/sys_site_language.php

Reply:
Rewrite is complete